### ISSUE TITLE
Docker got stuck on starting.

### ISSUE THREAD

**Issue by AndreyKrupskii** [docker/for-mac#3826]
### Summary

  - [x] I have tried with the latest version of my channel (Stable).
  - [x] I have uploaded Diagnostics.
  - Diagnostics ID: 27B6E148-025E-491B-8348-94FBD893D76B/20190809112334

### Expected behavior

Docker starts properly.

### Actual behavior

Docker got stuck on starting.

### Information

During issue investigation next points were observed:
- the issue reproduces constantly;
- the issue appeared after latest OSX - 10.14.6 (18G87) and Docker Desktop - 2.1.0.1 (37199) update.

## Resolution

Next resolution was found:
- `screen ~/Library/Containers/com.docker.docker/Data/vms/0/tty`;
- wait loading, select continue booting and press enter.

**Comment by Integralist** [comment]
I had the same issue.

Had to uninstall and reinstall and old version to unblock myself (e.g. get work done!)

```
$ brew cask install https://raw.githubusercontent.com/ahousseini/homebrew-cask/3d8ed99786fa59b9830f524ef3b1151e6098eedb/Casks/docker.rb

$ docker -v
Docker version 18.09.2, build 6247962
```

But that failed after first run of a docker container.

Then tried manually installing binary version `2.0.0.3`...

https://docs.docker.com/docker-for-mac/release-notes/#docker-community-edition-17120-ce-mac49-2018-01-19

This worked for me.

**Comment by Integralist** [comment]
Note: I had to restart my laptop and then the version `2.0.0.3` now crashes after starting up.

Also I can't grab diagnostics as the app crashes before the app generates anything.

**Comment by Integralist** [comment]
I tried to manually install the latest stable this morning, but that (as expected) crashes while docker tries to start.

I've now opted to install a non-stable (i.e. latest edge version -- as of yesterday 12th Aug `2.1.1.0`) and it looks to be running! 

![PRAISE JESUS](https://media1.giphy.com/media/vsCMqFTC9QugU/giphy.gif?cid=6104955ead59b8df5e986a6b92978fa727b07b2eef488bb4&rid=giphy.gif)

But I'll have to wait to see for _how long_ it stays running 😭

**Comment by j7an** [comment]
@AndreyKrupskii 's work around seems to work. Hopefully there's an update soon to the stable channel.

**Comment by mefril** [comment]
Have the same issue, diagnosis id 228F6194-79D8-4D46-BED7-FAE6E8489ECD/20190818182215

**Comment by dsandrade** [comment]
I have the same issue, and the screen don't work for me

**Comment by dsandrade** [comment]
diagnosis id is B5E12CE5-7DCB-40D5-B194-2B5A11EFB2FD/20190916233926

**Comment by TechnoTaff** [comment]
I still have this issue with the latest release...

Docker version 19.03.2, build 6a30dfc
MacOS version 10.14.6 (18G95)

Fatal error
The operation couldn’t be completed. (NIO.SocketAddressError error 3.)

**Comment by zhonghuiwen** [comment]
same issue, diagnosis id is FA1F46D3-FFF2-404F-8A05-905B14AB3F1E/20190924041124

**Comment by deepakcpakhale** [comment]
Thank you..!  @AndreyKrupskii It worked for me. Do you mind explaining why docker daemon remains stuck in the "starting" state and how "screen" helps it to bring back to "Running" state.

**Comment by kmblake** [comment]
I also observed this on with docker 19.02.3 on MacOS 10.14.6 (18G103). Using the screen command mentioned here worked. Also, after upgrading docker from stable to edge (19.03.3), this was no longer an issue.

**Comment by gbmhunter** [comment]
This occurred for me using docker version `19.03.2` (one revision less than the reported version above of `19.03.3` that seems to fix this issue. @AndreyKrupskii 's workaround fixed it for me also.

**Comment by xombified23** [comment]
Works! Thanks, @AndreyKrupskii !

**Comment by cmgsplk** [comment]
For those not understanding the work around, 

1. screen ~/Library/Containers/com.docker.docker/Data/vms/0/tty attaches to the tty for the VM
2. Selecting Continue (it's stuck on a window) lets teh vM keep booting.

I don't know exactly what the screen is saying because it was only a partial refresh but i typed continue then hit enter and it finished booting

**Comment by l3ender** [comment]
Seeing the same issue on OSX 10.15.1 (19B2106) and Docker Desktop 2.1.0.5 (40693). If I go to Troubleshooting, it never completing "diagnosing".

I tried rebooting, no help. Opening tty via `screen` results in a blank page and I can't do anything from there.

**Comment by ryanissamson** [comment]
> Seeing the same issue on OSX 10.15.1 (19B2106) and Docker Desktop 2.1.0.5 (40693). If I go to Troubleshooting, it never completing "diagnosing".
> 
> I tried rebooting, no help. Opening tty via `screen` results in a blank page and I can't do anything from there.

Apparently you have to wait a few minutes to not interrupt it. If I do it too early, I have to restart Docker again.

**Comment by ryanissamson** [comment]
Duplicate of #3390
